app/src/main/kotlin/io/intrepid/skotlinton/base/BaseViewModel.kt

```diff
@@ -15,9 +17,18 @@ open class BaseViewModel(configuration: ViewModelConfiguration) : ViewModel() {
     protected val timeProvider = configuration.timeProvider
     protected val crashReporter = configuration.crashReporter
 
-    protected val networkDisposables = CompositeDisposable()
+    protected val disposables = CompositeDisposable()
 
-    fun <T> Observable<T>.subscribeOnIoObserveOnUi(): Observable<T> = applySchedulers(ioScheduler, uiScheduler)
+    val eventPublisher: PublishRelay<ViewEvent> = PublishRelay.create()
```

Comment:
Should this have lower visibility? Seems strange for `sendViewEvent()` to be protected when `eventPublisher` is publicly accessibile

Reply:
this is a similar case with LiveData where this should be broken into a public Observable and a private PublishRelay, though it does make more sense to do so here.
In either case, the whole event communication mechanism needs more thoughts and additional work. I just created the `sendViewEvent()` and `onViewEvent()` so that it would be easy to change the underlying mechanism.